extensions/src/main/java/org/pentaho/platform/plugin/services/email/EmailService.java

```diff
@@ -147,12 +191,25 @@ public String sendEmailTest( final IEmailConfiguration emailConfig ) {
     emailProperties.setProperty( "mail.smtp.port", ObjectUtils.toString( emailConfig.getSmtpPort() ) );
     emailProperties.setProperty( "mail.transport.protocol", emailConfig.getSmtpProtocol() );
     emailProperties.setProperty( "mail.smtp.starttls.enable", ObjectUtils.toString( emailConfig.isUseStartTls() ) );
-    emailProperties.setProperty( "mail.smtp.auth", ObjectUtils.toString( emailConfig.isAuthenticate() ) );
     emailProperties.setProperty( "mail.smtp.ssl", ObjectUtils.toString( emailConfig.isUseSsl() ) );
     emailProperties.setProperty( "mail.debug", ObjectUtils.toString( emailConfig.isDebug() ) );
 
     Session session = null;
-    if ( emailConfig.isAuthenticate() ) {
+    IEmailAuthenticationResponse token = null;
+    if ( emailConfig.getAuthMechanism().equals( XOAUTH ) ) {
+      try {
+        token = getOAuthToken( emailConfig );
+        if ( emailConfig.getGrantType().equals( AUTHORIZATION_CODE  ) ) {
+          emailConfig.setGrantType( REFRESH_TOKEN );
+          emailConfig.setRefreshToken( token.getRefreshToken() );
+          emailConfig.setAuthorizationCode( "" );
+        }
+        session = Session.getInstance( emailProperties );
+      } catch ( Exception e ) {
+        return TEST_EMAIL_FAIL;
+      }
+    } else if ( emailConfig.isAuthenticate() ) {
+      emailProperties.setProperty( "mail.smtp.auth", "true" );
```

Comment:
why do you have to hard set this property here `"mail.smtp.auth"` to _"true"_?
In what scenario would this be false? In the baseline it was possible for it to be false no?